### PR TITLE
[Login] Fix: 'Revoked' auth flow handles user has no memberships

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
Cause: we got monitors with panic flag "unreachable: user has no memberships.", see [here](https://app.datadoghq.eu/logs?query=%22unreachable%3A%20user%20has%20no%20memberships.%22%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1716888517241&to_ts=1718184517241&live=true)

If a user tries to join a pro plan workspace via the regular flow (not invite) with a given target workspace id, doesn't have any existing membership and:
- has a non-verified email OR
- there's no workspace with verified domain matching their email OR
- a domain exist but it does not match the specified wId

the user is redirected to the /no-workspace page and the above panic flag is triggered and user-facing.

This PR avoids the situation by displaying the appropriate error to the user upfront.

Risk & Deploy
---
N/A

### CODE REVIEW

File: front/lib/iam/errors.ts [hunk 1]
```diff
@@ -9,3 +9,15 @@ export class AuthFlowError extends Error {
     super(message);
   }
 }
+
+export class UnverifiedEmailError extends Error {
```
Comment (flvndvd) [inline front/lib/iam/errors.ts:13]: We should avoid creating too many variation of errors if they don't contain more logic. What about using `AuthFlowError` for our flow?

**Comment by flvndvd** [comment]
> If a user tries to join a pro plan workspace via the regular flow (not invite) with a given target workspace id
How can this happen? IIRC, `targetWorkspaceId` comes from the url of the user.

**Comment by philipperolet** [comment]
What if a user was shared a join link but goes to it with an email that's not the company mail for autojoin ?  (verified or not)

**Comment by philipperolet** [comment]
Fixed in another pr cc @flvndvd

**Pull request closed without merging**